Tribler/Test/Core/CreditMining/test_creditmining_sys.py

```diff
@@ -0,0 +1,376 @@
+from Tribler.dispersy.endpoint import ManualEnpoint
+from Tribler.dispersy.member import DummyMember
+from Tribler.dispersy.util import blocking_call_on_reactor_thread
+
+FILE_DIR = os.path.abspath(os.path.dirname(os.path.realpath(__file__)))
+TORRENT_FILE = os.path.join(TESTS_DATA_DIR, "ubuntu-15.04-desktop-amd64.iso.torrent")
+TORRENT_FILE_INFOHASH = binascii.unhexlify("fc8a15a2faf2734dbb1dc5f7afdc5c9beaeb1f59")
+
+class TestBoostingManagerSys(TestAsServer):
+    """
+    base class to test base credit mining function
+    """
+
+    def __init__(self, *argv, **kwargs):
+        super(TestBoostingManagerSys, self).__init__(*argv, **kwargs)
+
+    def setUp(self, autoload_discovery=True):
+        super(TestBoostingManagerSys, self).setUp()
+
+        self.set_boosting_settings()
+        self.boosting_manager = BoostingManager.get_instance(self.session, self.bsettings)
+
+    def set_boosting_settings(self):
+        self.bsettings = BoostingSettings(self.session)
+        self.bsettings.credit_mining_path = self.session_base_dir
+        self.bsettings.config_file = 'None'
+        self.bsettings.min_connection_start = -1
+        self.bsettings.min_channels_start = -1
+
+        self.bsettings.max_torrents_active = 8
+        self.bsettings.max_torrents_per_source = 5
+        # self.bsettings.source_interval = 3
+        # self.bsettings.swarm_interval = 5
+        # self.bsettings.initial_swarm_interval = 1
+        self.bsettings.tracker_interval = 5
+        self.bsettings.initial_tracker_interval = 5
+        self.bsettings.logging_interval = 30
+        self.bsettings.initial_logging_interval = 3
+
+    def setUpPreSession(self):
+        super(TestBoostingManagerSys, self).setUpPreSession()
+
+        self.config.set_torrent_checking(True)
+        self.config.set_megacache(True)
+        self.config.set_dispersy(True)
+        self.config.set_mainline_dht(True)
+        self.config.set_torrent_store(True)
+        self.config.set_enable_torrent_search(True)
+        self.config.set_enable_channel_search(True)
+        # self.config.set_torrent_collecting(True)
+        self.config.set_libtorrent(True)
+        self.config.set_dht_torrent_collecting(True)
+        # self.config.set_enable_metadata(True)
+        # self.config.set_http_api_enabled(True)
+        # self.config.set_upgrader_enabled(False)
+
+    def tearDown(self):
+        self.boosting_manager.shutdown()
+        self.boosting_manager.del_instance()
+        os.remove('None')
+
+        super(TestBoostingManagerSys, self).tearDown()
+
+class TestBoostingManagerSysRSS(TestBoostingManagerSys):
+    """
+    testing class for RSS (dummy) source
+    """
+    def __init__(self, *argv, **kwargs):
+        super(TestBoostingManagerSysRSS, self).__init__(*argv, **kwargs)
+
+    def setUp(self, autoload_discovery=True):
+        super(TestBoostingManagerSysRSS, self).setUp()
+
+        # Prepare test_rss.xml file, replace the port with a random one
```

Comment:
This method is also used in one of the GUI tests. Can you refactor it to a separate method (something like `prepare_rss_feed(rss_feed)`) so we only have one implementation of this?